### ISSUE TITLE
Investigate APNG straight alpha

### ISSUE THREAD

**Issue by jtmcdole** [flutter/flutter#151506]
#151210 pop'd up as an issue with unpremultiplied alpha. It turns out APNG is converting frames to and from straight/premultiplied formats.

Maybe: just convert every frame to premultiplied and leave it?

**Comment by jonahwilliams** [comment]
Yes please. The per-frame conversion is going to be expensive if the image is huge.

**Comment by jtmcdole** [comment]
I'll spend part of today looking at this

**Comment by jtmcdole** [comment]
<back from vacation>

**Comment by gaaclarke** [comment]
I'm uncertain what the reproduction steps are here.  Is this closed after https://github.com/flutter/engine/pull/53770?
